libdnf5/base/goal.cpp

```diff
@@ -2313,15 +2314,33 @@ base::Transaction Goal::resolve() {
     sack->p_impl->make_provides_ready();
 
 
-    // TODO(jmracek) Apply modules first
     module::ModuleSack & module_sack = p_impl->base->module_sack;
     ret |= p_impl->add_module_specs_to_goal(transaction);
     // Resolve modules
     auto module_error = module_sack.resolve_active_module_items().second;
-    if (module_error != libdnf5::module::ModuleSack::ModuleErrorType::NO_ERROR) {
-        throw module::ModuleResolveError(M_("Failed to resolve modules."));
+
+    // Report problems from the module solver
+    auto module_solver_problems =
+        base::process_module_solver_problems(module_sack.p_impl->pool, module_sack.p_impl->solver_problems);
+    auto module_goal_problem = module_sack.p_impl->module_solver_error_to_goal_problem(module_error);
```

Comment:
This converter is somehow additional step, but removal  of it would result in modification of enums, changes on API. From my point of view it would require much more energy then benefits.